app/src/main/java/org/mate/accessibility/check/widgetbased/AcionableElementsCheck.java

```diff
@@ -1,12 +1,44 @@
 package org.mate.accessibility.check.widgetbased;
 
+import org.mate.MATE;
+import org.mate.Registry;
 import org.mate.accessibility.AccessibilityViolation;
+import org.mate.accessibility.AccessibilityViolationTypes;
 import org.mate.state.IScreenState;
+import org.mate.ui.EnvironmentManager;
 import org.mate.ui.Widget;
 
 public class AcionableElementsCheck implements IWidgetAccessibilityCheck {
     @Override
     public AccessibilityViolation check(IScreenState state, Widget widget) {
+
+        if (!widget.isImportantForAccessibility())
+            return null;
+
+        boolean checkClickableText=false;
+        if (widget.getClazz().equals("android.widget.TextView") && widget.isClickable() && !widget.isEditable() && !widget.mightBeImage()){
+            checkClickableText=true;
+            //return new AccessibilityViolation(AccessibilityViolationTypes.ACTIONABLE_ELEMENTS,widget,state,"Clickable text, not button");
+        }
+
+        //if (check whether background color of the button is the same as the screen background)
+        double matchesBackgroundColor = 0;
+
+        try{ matchesBackgroundColor = Registry.getEnvironmentManager().matchesSurroundingColor(state.getPackageName(),state.getId(),widget);}
```

Comment:
Should not use try ... catch here as the called method does not throw a checked exception and any runtime exceptions like loosing connection to the MATE-server should not be handled here.

Reply:
indeed